java/client/src/test/java/glide/utils/RustCoreMock.java

```diff
@@ -0,0 +1,187 @@
+    }
+
+    @SneakyThrows
+    private RustCoreMock() {
+        var threadPoolResource = Platform.getThreadPoolResourceSupplier().get();
+        socketPath = Files.createTempFile("GlideCoreMock", null).toString();
+        group = threadPoolResource.getEventLoopGroup();
+        channel =
+                new ServerBootstrap()
+                        .group(group)
+                        .channel(
+                                Platform.getCapabilities().isEPollAvailable()
+                                        ? EpollServerDomainSocketChannel.class
+                                        : KQueueServerDomainSocketChannel.class)
+                        .childHandler(
+                                new ChannelInitializer<DomainSocketChannel>() {
+
+                                    @Override
+                                    protected void initChannel(DomainSocketChannel ch) throws Exception {
+                                        ch.pipeline()
+                                                // https://netty.io/4.1/api/io/netty/handler/codec/protobuf/ProtobufEncoder.html
+                                                .addLast("frameDecoder", new ProtobufVarint32FrameDecoder())
+                                                .addLast("frameEncoder", new ProtobufVarint32LengthFieldPrepender())
+                                                .addLast("protobufEncoder", new ProtobufEncoder())
+                                                .addLast(
+                                                        new ChannelInboundHandlerAdapter() {
+
+                                                            // This works with only one connected client.
+                                                            // TODO Rework with `channelActive` override.
+                                                            private final AtomicBoolean anybodyConnected =
+                                                                    new AtomicBoolean(false);
+
+                                                            @Override
+                                                            public void channelRead(ChannelHandlerContext ctx, Object msg)
+                                                                    throws Exception {
+                                                                var buf = (ByteBuf) msg;
+                                                                var bytes = new byte[buf.readableBytes()];
+                                                                buf.readBytes(bytes);
+                                                                buf.release();
+                                                                if (messageProcessor.isRaw()) {
+                                                                    ch.writeAndFlush(
+                                                                            Unpooled.copiedBuffer(messageProcessor.handle(bytes)));
```

Comment:
This has alot of nesting. Anyway it can be reduced? It's deep enough that it's pretty unreadable in GitHub's diff view.

Reply:
@Yury-Fridlyand at least create a separate class for ChannelInboundHandlerAdapter?